src/connections.rs

```diff
@@ -19,19 +20,18 @@ use std::{net::SocketAddr, sync::Arc};
 use tokio::select;
 
 /// Connection instance to a node which can be used to send messages to it
+#[derive(Clone)]
 pub struct Connection {
     quic_conn: quinn::Connection,
-}
-
-impl Drop for Connection {
-    fn drop(&mut self) {
-        self.close();
-    }
+    pool_handle: ConnectionHandle,
```

Comment:
Shouldn't we call it `conn_handle` instead as we are not really interested in the pool here but just the connection we got the handle for?

Reply:
Well it's purpose is to remove the connection from the pool, so having `pool` in the name makes sense in my opinion. But as said in the previous comment, maybe the name isn't clear?

Reply:
It's not a big deal, but then why the struct is named ConnectionHandle, whatever that struct is called you are not getting a handle to the pool, but a handle to somethings that knows how to remove the connection from the pool, so that's what I mean.